tests.cpp

```diff
@@ -2,52 +2,203 @@
     std::vector<std::unique_ptr<ElectricEngine>> OldEngineShop;
 
-    REQUIRE(Tesla.getElectricEngineData()->getBatteryCapacity() == 4000);
-    REQUIRE(Tesla.getElectricEngineData()->getPower() >= 600);
+    REQUIRE(Tesla.getElectricEngineData()->getBatteryCapacity() ==
+            batteryCapacity1);
+    REQUIRE(Tesla.getElectricEngineData()->getPower() >= power1);
 
     WHEN("car is moving") {
-      Tesla.accelerate(50);
 
+      Tesla.accelerate(speed);
       THEN("Cannot pull out engine") {
         REQUIRE_THROWS_AS(Tesla.pullOutElectricEngine(), std::runtime_error);
-        REQUIRE(Tesla.getElectricEngineData()->getBatteryCapacity() == 4000);
-        REQUIRE(Tesla.getElectricEngineData()->getPower() >= 600);
+        REQUIRE(Tesla.getElectricEngineData()->getBatteryCapacity() ==
+                batteryCapacity1);
+        REQUIRE(Tesla.getElectricEngineData()->getPower() >= power1);
         REQUIRE(OldEngineShop.size() == 0);
       }
     }
     WHEN("car has stopped") {
-      Tesla.accelerate(0);
+      Tesla.accelerate(stopSpeed);
       THEN("returned engine unique ptr is not nullptr") {
         REQUIRE(Tesla.pullOutElectricEngine() != nullptr);
       }
     }
 
     WHEN("car has stopped and pulled out engine") {
-      Tesla.accelerate(0);
+      Tesla.accelerate(stopSpeed);
       Tesla.pullOutElectricEngine();
       THEN("unique_ptr is nullptr") {
         REQUIRE(Tesla.pullOutElectricEngine() == nullptr);
       }
     }
     WHEN("car has stopped and put in another engine") {
-      Tesla.accelerate(0);
+      Tesla.accelerate(stopSpeed);
       OldEngineShop.push_back(Tesla.pullOutElectricEngine());
-      Tesla.putInElectricEngine(std::make_unique<ElectricEngine>(400, 3000));
+      Tesla.putInElectricEngine(
+          std::make_unique<ElectricEngine>(power2, batteryCapacity2));
 
       THEN("Check if there are new parameters") {
-        REQUIRE(Tesla.getElectricEngineData()->getBatteryCapacity() == 3000);
-        REQUIRE(Tesla.getElectricEngineData()->getPower() >= 400);
+        REQUIRE(Tesla.getElectricEngineData()->getBatteryCapacity() ==
+                batteryCapacity2);
+        REQUIRE(Tesla.getElectricEngineData()->getPower() >= power2);
       }
     }
   }
 }
+
+SCENARIO("Checking if Petrol functions works", "[PetrolCar]") {
+
+  GIVEN("PetrolCar and Shop") {
+    int power1 = 600;
+    int power2 = 400;
+    int gear_number1 = 5;
+    int gear_number2 = 6;
+    int cubicCapacity1 = 1600;
+    int cubicCapacity2 = 2000;
+    int stopSpeed = 0;
+    int speed = 50;
+    PetrolCar Golf(
+        std::make_unique<PetrolEngine>(power1, cubicCapacity1, gear_number1));
+    std::vector<std::unique_ptr<PetrolEngine>> OldEngineShop;
+
+    REQUIRE(Golf.getPetrolEngineData()->getCapacity() == cubicCapacity1);
+    REQUIRE(Golf.getPetrolEngineData()->getPower() >= power1);
+
+    WHEN("car is moving, and checking car data") {
+      int ActualGear = 3;
```

Comment:
you mix UpperCamelCase with lowerCamelCase in variable names. Please use uniform way.